cmd/topicctl/subcmd/apply.go

```diff
@@ -320,15 +325,18 @@ func applyTopic(
 		if err != nil {
 			// if one of the steps after updateSettings errors when updating a topic,
 			// we can be in a state where some (but not all) changes were applied
+			// some topic creation errors also still create the topic
 			changesToBePrinted = unpackChanges(topicChanges, changesToBePrinted)
-			log.Error("Error detected while updating topic, the following changes were still made:")
-			partialChanges, err := printChanges(changesToBePrinted)
-			if err != nil {
-				log.Error("Error printing JSON changes data")
+			if changesToBePrinted.NewTopics != nil || changesToBePrinted.UpdatedTopics != nil {
```

Comment:
it's possible to get an error during topic creation but still have the topic be created, such as if the topic's placement strategy can't be fulfilled (i.e. `in-rack` strategy but there's not enough racks for the given replication factor)